src/main.rs

```diff
@@ -1,10 +1,5 @@
 /// floki - the development container launcher
-#[macro_use]
-extern crate quicli;
-#[macro_use]
-extern crate failure;
-extern crate serde_yaml;
-extern crate uuid;
+#[macro_use] extern crate failure_derive;
```

Comment:
I'm not really clear on the change that happened in Rust 2018 that makes this possible - can you explain for my education?

Reply:
Sure - take a look at this: https://doc.rust-lang.org/edition-guide/rust-2018/module-system/path-clarity.html#no-more-extern-crate

And here's a summary, from the original blog post: https://blog.rust-lang.org/2018/12/06/Rust-1.31-and-rust-2018.html#module-system-changes

Reply:
You don't even need to `extern crate` for macros any more, you can write

```
#[derive(failure::Fail)]
struct FooError(String);
```

Reply:
Thanks - that makes complete sense. Always wondered why extern crate was really necessary...

Will use this knowledge from now on!

Reply:
Thanks - that makes complete sense. Always wondered why extern crate was really necessary...

Will use this knowledge from now on!